sycl/include/sycl/ext/intel/experimental/fpga_utils.hpp

```diff
@@ -19,85 +20,43 @@ namespace ext {
 namespace intel {
 namespace experimental {
 
-enum class type {
-  none, // default
-  exact,
-  max,
-  min
-};
-
-template <int32_t _N> struct latency_anchor_id {
-  static constexpr int32_t value = _N;
-  static constexpr int32_t default_value = -1;
-};
-
-template <int32_t _N1, type _N2, int32_t _N3> struct latency_constraint {
-  static constexpr std::tuple<int32_t, type, int32_t> value = {_N1, _N2, _N3};
-  static constexpr std::tuple<int32_t, type, int32_t> default_value = {
-      0, type::none, 0};
-};
-
-using ignoreParam_int_t = int32_t;
-constexpr ignoreParam_int_t IgnoreParamInt{};
-using ignoreParam_enum_t = type;
-constexpr ignoreParam_enum_t IgnoreParamEnum{};
-
-template <class _VType, class _T> struct _ValueExtractorImp {
-  static constexpr auto _First = _T::value;
-  static constexpr auto _Second = IgnoreParamEnum;
-  static constexpr auto _Third = IgnoreParamInt;
-};
+template <template <int32_t> class _Type, class _T>
+struct _MatchType : std::is_same<_Type<_T::value>, _T> {};
```

Comment:
How does the `detail` namespace help avoid the confusion here?

Reply:
`detail` is used for implementation detail constructs and as such a user should not use them. Though you roughly have the same here with `_` prefixing some of the structs, I think it would be better to move them into `detail` so users do not get the wrong idea.

Reply:
If a "user" here means outside this header file, then sure I can move `_MatchType` into the `detail` namespace. `_GetValue` and `GetOrDefaultValT` are used elsewhere, so I don't think I should touch them.

Reply:
In general, the only public symbols in our headers defined outside of the `detail` namespace should be the public API that is described in the extension spec.  The reason is that customers sometimes look at the headers and just assume that they can use any symbol they see as a public API.  Then, we end up needing to support these symbols in order to avoid breaking customer code.

I assume `_GetValue` and `GetOrDefaultValT` are not part of the public API.  Why can't they be moved to `detail`?

Reply:
Okay, if "users" in this context means external customers, then everything in fpga_utils.hpp should be moved into the `detail` namespace.